src/components/Search/SearchDropdown.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
+import Grid from '@mui/material/Grid';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import SearchViewHelper from '../../utils/searchViewHelper';
+
+const { handleChangeDropdown } = SearchViewHelper;
+
+function SearchInput(props) {
+  const { dropdownError, dropdownHelperText, repository } = props;
+  const repoList = ['try', 'autoland', 'mozilla-central'];
+
+  return (
+    <Grid item xs={2}>
+      {!dropdownError && !dropdownHelperText ? (
+        /* dropdown without errors */
+        <FormControl sx={{ width: '100%' }}>
+          <InputLabel id="select-repository">repository</InputLabel>
+          <Select
+            value={repository}
+            labelId="select-repository"
+            label="repository"
+          >
+            {repoList.length > 0 &&
+              repoList.map((item) => (
+                <MenuItem
+                  id={item}
+                  value={item}
+                  key={item}
+                  onClick={handleChangeDropdown}
+                >
+                  {item}
+                </MenuItem>
+              ))}
+          </Select>
+        </FormControl>
+      ) : (
+        /* dropdown with errors */
+        <FormControl error sx={{ width: '100%' }}>
+          <InputLabel id="select-repository">repository</InputLabel>
+          <Select
+            value={repository}
+            labelId="select-repository"
+            label="repository"
+          >
+            {repoList.length > 0 &&
+              repoList.map((item) => (
+                <MenuItem
+                  id={item}
+                  value={item}
+                  key={item}
+                  onClick={handleChangeDropdown}
+                >
+                  {item}
+                </MenuItem>
+              ))}
+          </Select>
+          <FormHelperText>{dropdownHelperText}</FormHelperText>
```

Comment:
Maybe I'm missing something, but it looks like this line is the only difference (+ the `error` prop). Therefore maybe you could add the condition for these places? This would help to visually see what the differences are between these 2 cases.

For example:
```js
<FormControl error={dropdownError} sx={{ width: '100%' }}>
  <InputLabel>...</InputLabel>
  <Select>...</Select>
  {dropdownHelperText ? <FormHelperText>{dropdownHelperText}</FormHelperText> : null}
</FormControl>
```

This would also make it easier in the future to show some text even when there's no error.